bigtable-client-core-parent/bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestBigtableOptionsFactory.java

```diff
@@ -176,4 +213,238 @@ public void testLongOperationsTimeouts() throws IOException {
     assertEquals(
         Integer.parseInt(readTimeout), options.getCallOptionsConfig().getReadStreamRpcTimeoutMs());
   }
+
+  @Test
+  public void testDataSettingsBasicKeys() throws IOException {
+    String appProfileId = "appProfileId";
+    String userAgent = "test-user-agent";
+    Credentials credentials = Mockito.mock(Credentials.class);
+
+    configuration.set(BigtableOptionsFactory.BIGTABLE_PORT_KEY, String.valueOf(TEST_PORT));
+    configuration.set(BigtableOptionsFactory.APP_PROFILE_ID_KEY, appProfileId);
+    configuration.setBoolean(BIGTABLE_USE_PLAINTEXT_NEGOTIATION, true);
+    configuration.set(BigtableOptionsFactory.CUSTOM_USER_AGENT_KEY, userAgent);
+    configuration = BigtableConfiguration.withCredentials(configuration, credentials);
+
+    BigtableDataSettings dataSettings =
+        BigtableOptionsFactory.createBigtableDataSettings(configuration);
+    assertEquals(TEST_PROJECT_ID, dataSettings.getProjectId());
+    assertEquals(TEST_INSTANCE_ID, dataSettings.getInstanceId());
+    assertEquals(appProfileId, dataSettings.getAppProfileId());
+
+    assertEquals(TEST_HOST + ":" + TEST_PORT, dataSettings.getStubSettings().getEndpoint());
+    Map<String, String> headers = dataSettings.getStubSettings().getHeaderProvider().getHeaders();
+    assertTrue(headers.get(GrpcUtil.USER_AGENT_KEY.name()).contains(userAgent));
+    assertEquals(
+        credentials, dataSettings.getStubSettings().getCredentialsProvider().getCredentials());
+  }
+
+  @Test
+  public void testAdminSettingsBasicKeys() throws IOException {
+    String adminHost = "testadmin.example.com";
+    String userAgent = "test-user-agent";
+    Credentials credentials = Mockito.mock(Credentials.class);
+
+    configuration.set(BigtableOptionsFactory.BIGTABLE_ADMIN_HOST_KEY, adminHost);
+    configuration.setInt(BigtableOptionsFactory.BIGTABLE_PORT_KEY, TEST_PORT);
+    configuration.set(BigtableOptionsFactory.CUSTOM_USER_AGENT_KEY, userAgent);
+    configuration.setBoolean(BIGTABLE_USE_PLAINTEXT_NEGOTIATION, true);
+    configuration = BigtableConfiguration.withCredentials(configuration, credentials);
+
```

Comment:
Here and below: you can add static import of other BigtableOptionsFactory properties